ertviz/controllers/ensemble_selector_controller.py

```diff
@@ -0,0 +1,88 @@
+import json
+import random
+from dash.dependencies import Input, Output, State
+from dash.exceptions import PreventUpdate
+from ertviz.data_loader import get_ensembles
+from ertviz.models import EnsembleModel
+
+
+def ensemble_selector_controller(parent, app):
+    @app.callback(
+        Output(parent.uuid("ensemble-selector"), "layout"),
+        [Input("dropdown-layout", "value")],
+    )
+    def update_cytoscape_layout(layout):
+        return {"name": layout}
+
+    @app.callback(
+        Output(parent.uuid("ensemble-selector"), "elements"), [Input("url", "pathname")]
+    )
+    def update_ensemble_selector(query):
+
+        ensemble_dict = get_ensembles()
+        for ensemble_schema in ensemble_dict:
+            ensemble_id = ensemble_schema["ref_url"]
+            if ensemble_id not in parent.ensembles:
+                parent.ensembles[ensemble_id] = EnsembleModel(ref_url=ensemble_id)
+
+        roots = [
+            ensemble
+            for ensemble_id, ensemble in parent.ensembles.items()
+            if ensemble.parent == None
+        ]
+
+        datas = []
+        queue = []
+        for index, ensemble in enumerate(roots):
+            queue.append(ensemble)
+
+        # coloring at random, should probably be more deterministic
+        r = lambda: random.randint(0, 255)
+
+        def _construct_node(ensemble):
+            return {
+                "data": {
+                    "id": ensemble.id,
+                    "label": ensemble._name,
+                    "color": f"rgb({r()},{r()},{r()})",
+                }
+            }
+
+        def _construct_edge(ensemble_src, ensemble_target):
+            return {
+                "data": {
+                    "source": ensemble_src.id,
+                    "target": ensemble_target.id,
+                }
+            }
+
+        while queue:
```

Comment:
I would encapsulate this to a function, which would take pandas df instead - to alleviate testing. Similarly to `def _get_realizations_plots` in `response_controller.py`

Reply:
This does not fit into a pandas frame, as it is relational data.

Reply:
I'll look at it again ... but the idea was to get the unit tests for the graph into `test_controller.py` What do you think

Reply:
A test would probably be good

Reply:
So you did the test, nice!